### DISCUSSION TITLE
Android app:  no download button for repo items

### DISCUSSION THREAD

**Discussion by 4evermaat** [community/community#36511] [Mobile]
I recently tried to perform a simple download of a file

https://github.com/anars/blank-audio/blob/master/5-seconds-of-silence.mp3

Within the android app, i could not do this on the repo; i kept getting shown the (binary) source code of the file.

[link](https://user-images.githubusercontent.com/830306/196338001-6422aa8f-d7c9-4091-a187-dfc2ca3559c8.mp4) 

I had to open the repo folder in chrome browser and manually select download option.

[](https://user-images.githubusercontent.com/830306/196338127-df5cd842-5404-4390-9567-60134c468b58.mp4) 

Forgive me if the video didnt embed.  I have no idea how to embed them; google search for instructions was fruitless.

**Comment by iskae** [comment]
Hello @4evermaat 👋 

Thanks so much for reaching out ❤️ 

You are right, we are currently not supporting downloading binary files to device, but this could be a nice addition to our apps 👍 

I am creating an internal issue for tracking this. We will evaluate it and hopefully get it to our clients real soon 😊 

Cheers!

**Reply by iskae** [reply]
Oh so sorry @4evermaat , replied too early and I was about to edit the comment with this information 🙇 

You can simply use the uploaded image as is, like:

```

<details><summary>Video</summary>

https://user-images.githubusercontent.com/830306/196338001-6422aa8f-d7c9-4091-a187-dfc2ca3559c8.mp4

</details>
```
<details><summary>Video</summary>

https://user-images.githubusercontent.com/830306/196338001-6422aa8f-d7c9-4091-a187-dfc2ca3559c8.mp4

</details>

Or you can create a link out of it, like:

```
[Link to the video](https://user-images.githubusercontent.com/830306/196338001-6422aa8f-d7c9-4091-a187-dfc2ca3559c8.mp4)
```

[Link to the video](https://user-images.githubusercontent.com/830306/196338001-6422aa8f-d7c9-4091-a187-dfc2ca3559c8.mp4)


I hope this helps 😊

**Reply by 4evermaat** [reply]
@iskae You have to have a blank line between the last text and the video embed. And perhaps a blank line afterwards in order for the embed to work properly

[](https://user-images.githubusercontent.com/830306/196431760-2529a6a8-112f-4788-b092-c4311f6ce3a7.mp4)

I didnt have the blank line in between. So it wasnt parsing correctly. Or it would post a hyperlinked text. But with the extra line break separating other text before/after the video url (encased as hyperlink), the video player parses correctly.

The github blog post [announcing to the world](https://github.blog/2021-05-13-video-uploads-available-github/) that they now accept videos in posts should be updated with a how to use in post section.
Or direct link to how-to article.

**Comment by 4evermaat** [comment]
@iskae or others, can you comment on video embedding here and in github issues posts?

My uploaded videos i was unable to embed the mp4 as a video that could be consumed by a visitor of this issue.  They would have to download it and play it offline.

Edit:  continue here:  https://github.com/community/community/discussions/36511#discussioncomment-3905602

**Comment by ossanna16** [comment]
Hey there @4evermaat and thank you for asking a good question! I'm glad you found our community 🙂
